### PR TITLE
Translated email subjects 

### PR DESCRIPTION
**Pull request by Baggerone**
Sorry, I had to put the post status transitions in alphabetical order.  I was struggling with getting things set up in the yaml localization file with the order I had them in originally.

### CODE REVIEW

File: application/listeners/posthelpers.go [hunk 1]
```diff
@@ -2,12 +2,29 @@ package listeners
...
 	"github.com/silinternational/wecarry-api/models"
 	"github.com/silinternational/wecarry-api/notifications"
 )
 
+// TODO create a proper solution for this
+func getUserLanguage() string {
+	return "en"
+}
+
+type argAppName struct{ AppName string }
```
Comment (Schparky) [inline application/listeners/posthelpers.go]: Will this work as a map[string]string as well?

File: application/listeners/posthelpers.go [hunk 2]
```diff
@@ -2,12 +2,29 @@ package listeners
...
+
+func getTranslatedSubject(translationID, template string) string {
+	subj, err := domain.TranslateWithLang(getUserLanguage(), translationID, argAppName{AppName: domain.Env.AppName})
```
Comment (Schparky) [inline application/listeners/posthelpers.go]: Would it make sense to include the `AppName` arg in the `TranslateWithLang` function? Might be an arg that is needed more often than not.

File: application/notifications/messages.go [hunk 3]
```diff
@@ -9,4 +9,5 @@ type Message struct {
 	ToName    string
 	ToEmail   string
 	ToPhone   string
+	Subject   string
```
Comment (Schparky) [inline application/notifications/messages.go]: This field isn't read anywhere. Looks like `(e *EmailNotifier) Send` function needs to pass it on to the email service.

File: application/domain/domain.go [hunk 4]
```diff
@@ -347,3 +352,21 @@ func GetPostUIURL(postUUID string) string {
 func GetThreadUIURL(threadUUID string) string {
 	return Env.UIURL + threadUIPath + threadUUID
 }
+
+// TODO watch for when this is available on a new version of mw-i18n and then replace calls to this function
+// TranslateWithLang returns the translation of the string identified by translationID, for the given language.
+// See Translate for further details.
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/domain/domain.go]: Hopefully, we won't need a separate function like this once they have the original released.

File: application/listeners/posthelpers.go (hunk 2 above)
Comment (Baggerone) [reply application/listeners/posthelpers.go]: Hopefully the `TranslateWithLang` function will be getting replaced by a near-future upgrade.

File: application/notifications/messages.go (hunk 3 above)
Comment (Baggerone) [reply application/notifications/messages.go]: Good catch.

**Pull request merged**